apps/job-service/src/jobs/delayed-jobs.controller.ts

```diff
@@ -1,38 +1,116 @@
   })
   @JsonApiResponse({ data: { type: DelayedJobDto } })
   @ApiException(() => UnauthorizedException, {
     description: 'Authentication failed.',
   })
   @ApiException(() => NotFoundException, {
-    description: 'Job with that UUID not found.'
+    description: 'Job with that UUID not found.',
   })
+  // Note: Since jobs are very generic and we don't track which resources are related to a given
+  // job, there is no effective way to make a policy for jobs until we expand the service to
+  // include an owner ID on the job table.
   async findOne(@Param('uuid') pathUUID: string): Promise<JsonApiDocument> {
-    const job = await DelayedJob.findOne({ where: { uuid: pathUUID }});
+    const job = await DelayedJob.findOne({ where: { uuid: pathUUID } });
     if (job == null) throw new NotFoundException();
 
-    // Note: Since jobs are very generic and we don't track which resources are related to a given
-    // job, there is no effective way to make a policy for jobs until we expand the service to
-    // include an owner ID on the job table.
-
     return buildJsonApi()
       .addData(pathUUID, new DelayedJobDto(job))
       .document.serialize();
   }
-}
+
+  @Patch('bulk-clear')
+  @ApiOperation({
+    operationId: 'bulkClearJobs',
+    summary: 'Bulk update jobs to modify isAcknowledged for specified job IDs',
+    description: `Accepts a JSON:API-compliant payload to bulk update jobs, allowing each job's isAcknowledged attribute to be set to true or false.`,
+  })
+  @JsonApiResponse({ data: { type: DelayedJobDto } })
+  @ApiException(() => UnauthorizedException, { description: 'Authentication failed.' })
+  @ApiException(() => BadRequestException, { description: 'Invalid payload or IDs provided.' })
+  @ApiException(() => NotFoundException, { description: 'One or more jobs specified in the payload could not be found.' })
+  async bulkClearJobs(
+    @Body() bulkClearJobsDto: DelayedJobBulkUpdateBodyDto,
+    @Request() { authenticatedUserId }
+  ): Promise<JsonApiDocument> {
+    const jobUpdates = bulkClearJobsDto.data;
+    const updatePromises = jobUpdates.map(async (job) => {
+      const [updatedCount] = await DelayedJob.update(
+        { isAcknowledged: job.attributes.isAcknowledged },
+        {
+          where: {
+            uuid: job.uuid,
+            createdBy: authenticatedUserId,
+            status: { [Op.ne]: 'pending' },
+          },
+        });
+    
+      if (updatedCount === 0) {
+        throw new NotFoundException(`Job with UUID ${job.uuid} could not be updated.`);
+      }
+    
+      const updatedJob = await DelayedJob.findOne({
+        where: { uuid: job.uuid },
+      });
+
+      return updatedJob;
+    });
+    
+    const updatedJobs = await Promise.all(updatePromises);
+    
+    
+    const jsonApiBuilder = buildJsonApi();
+    updatedJobs.forEach((job) => {
+      jsonApiBuilder.addData(job.uuid, new DelayedJobDto(job));
+    });
+    
+    return jsonApiBuilder.serialize();
+    
```

Comment:
:art: It's a minor thing, but these extra blank lines are a little annoying. If you set up your editor to use the prettier config on save, it'll take care of stuff like that for you.

Reply:
great, that prettier will be helpfull. Changed